Makefile

```diff
@@ -301,90 +314,89 @@ endif
 docker-login:
 	@docker login -u $(DOCKER_USER) -p $(DOCKER_PASSWORD) $(REGISTRY)
 
-# Tag for Dev
-docker-tag-dev:
-	@docker tag $(IMG) $(REPOSITORY):$(DEV_TAG)
-	@docker tag $(IMG) $(REPOSITORY):dev
-	@docker tag $(CRD_IMG) $(CRD_REPOSITORY):$(DEV_TAG)
-	@docker tag $(CRD_IMG) $(CRD_REPOSITORY):dev
-	@docker tag $(GATOR_IMG) $(GATOR_REPOSITORY):$(DEV_TAG)
-	@docker tag $(GATOR_IMG) $(GATOR_REPOSITORY):dev
-
-# Tag for Dev
-docker-tag-release:
-	@docker tag $(IMG) $(REPOSITORY):$(VERSION)
-	@docker tag $(CRD_IMG) $(CRD_REPOSITORY):$(VERSION)
-	@docker tag $(GATOR_IMG) $(GATOR_REPOSITORY):$(VERSION)
-
-# Push for Dev
-docker-push-dev: docker-tag-dev
-	@docker push $(REPOSITORY):$(DEV_TAG)
-	@docker push $(REPOSITORY):dev
-	@docker push $(CRD_REPOSITORY):$(DEV_TAG)
-	@docker push $(CRD_REPOSITORY):dev
-	@docker push $(GATOR_REPOSITORY):$(DEV_TAG)
-	@docker push $(GATOR_REPOSITORY):dev
-
-# Push for Release
-docker-push-release: docker-tag-release
-	@docker push $(REPOSITORY):$(VERSION)
-	@docker push $(CRD_REPOSITORY):$(VERSION)
-	@docker push $(GATOR_REPOSITORY):$(VERSION)
-
-# Add crds to gatekeeper-crds image
-# Build gatekeeper image
-docker-build: build-crds
```

Comment:
Can we keep docker-build for backwards compatibility? Or does this break with the Dockerfile change?

Reply:
I removed it since cluttering the makefile and it's dead code, it's not used anywhere.

Reply:
I'd be a bit careful about calling makefile rules dead code, since they are meant to be how developers interact with the project (and therefore might be used even if not referenced), for instance, this is my go-to rule for building the image.

It looks like the `buildx` flavor is working for me, so I have no personal objections to it going away, but unsure if there are other pipelines out there.

Reply:
we don't document the `make docker-build` in our dev docs: https://open-policy-agent.github.io/gatekeeper/website/docs/install#deploying-head-using-make.

We do have `make docker-push` i'll update this doc in this pr